### PR TITLE
NXP-29499: fix error management on invalid operation/chain/codec (lazy check on chains)

### PR DESCRIPTION
**Pull request by atchertchian**
T&P check at https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-atchertchian-master-3/13/

### CODE REVIEW

**Comment by atchertchian** [comment]
Changed the strategy compared to #4239 to avoid to many changes: requirement is not needed anymore with this check at startup.
Note the important change here is that the chain is registered to the service, while this was prevented with a direct check (but in strict mode, startup will still fail).

File: modules/platform/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/AutomationComponent.java [hunk 1]
```diff
@@ -135,21 +136,25 @@ public void registerContribution(Object contribution, String extensionPoint, Com
...
             }
+            Class<?> type;
             try {
-                service.putOperation(opc.type, opc.replace, contributor.getName().toString(), widgetDefinitionList);
+                type = Class.forName(opc.type);
+            } catch (ClassNotFoundException e) {
+                throw new IllegalArgumentException("Invalid operation class '" + opc.type + "': class not found.");
+            }
+            try {
+                service.putOperation(type, opc.replace, contributor.getName().toString(), widgetDefinitionList);
```
Comment (kevinleturc) [inline modules/platform/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/AutomationComponent.java:146]: What do you think about merging the two try/catch? Something like below:

```java
            try {
                Class<?> type = Class.forName(opc.type);
                service.putOperation(type, opc.replace, contributor.getName().toString(), widgetDefinitionList);
            } catch (ClassNotFoundException e) {
                throw new IllegalArgumentException("Invalid operation class '" + opc.type + "': class not found.");
            } catch (OperationException e) {
                throw new RuntimeException(e);
            }
```

Comment (atchertchian) [reply modules/platform/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/AutomationComponent.java:146]: will let the T&P go through like this first, and will make the change after

**Comment by atchertchian** [comment]
Rebased to handle above feedback + reworked the last commit message

File: modules/platform/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/AutomationComponent.java [hunk 2]
```diff
@@ -135,21 +136,23 @@ public void registerContribution(Object contribution, String extensionPoint, Com
                     widgetDefinitionList.add(widgetDescriptor.getWidgetDefinition());
                 }
             }
+            Class<?> type;
```
Comment (kevinleturc) [inline modules/platform/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/AutomationComponent.java:139]: Not sure you need this declaration outside the try/catch?

Comment (atchertchian) [reply modules/platform/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/AutomationComponent.java:139]: indeed

**Comment by atchertchian** [comment]
Rebasing to replace "since 11.2" references by "since 11.3" because of release, and ignoring misleading "pending" statuses linked to changes on the checks.

**Pull request merged**